cppcache/include/geode/Execution.hpp

```diff
@@ -17,100 +17,122 @@
 
 #pragma once
 
-#ifndef GEODE_EXECUTION_H_
-#define GEODE_EXECUTION_H_
+#ifndef GEODE_EXECUTIONIMPL_H_
+#define GEODE_EXECUTIONIMPL_H_
 
-#include <memory>
-#include <chrono>
-#include <string>
+#include <map>
 
-#include "internal/geode_globals.hpp"
-#include "CacheableBuiltins.hpp"
-#include "ResultCollector.hpp"
+#include <ace/Condition_Recursive_Thread_Mutex.h>
+#include <ace/Guard_T.h>
```

Comment:
I don't think we want to leak out this ACE dependency in a public header.  However, if I recall correctly you may be able to replace these usages with `std::lock_guard` and `std::mutex` or `std::recursive_mutex`.

Reply:
Yeah and then the C++/CLI will fail to compile because `std::mutex` is forbidden. 

This attemp will need to be refactored into a modern pimpl pattern.